django/db/models/fields/__init__.py

```diff
@@ -761,11 +761,7 @@ def contribute_to_class(self, cls, name, private_only=False):
         self.model = cls
         cls._meta.add_field(self, private=private_only)
         if self.column:
-            # Don't override classmethods with the descriptor. This means that
-            # if you have a classmethod and a field with the same name, then
-            # such fields can't be deferred (we don't have a check for this).
-            if not getattr(cls, self.attname, None):
-                setattr(cls, self.attname, self.descriptor_class(self))
+            setattr(cls, self.attname, self.descriptor_class(self))
```

Comment:
We could:
- delay merging this work, and plop a warning/deprecation message in here instead, preparing people for the new behaviour. Arguably the urgency of the fix outweighs the niche change in field inheritance interaction.
- raise a warning (or bubble up into a django check) that the user has set an attribute with the same name twice and whatever they have on their parent is getting ignored.

However I think both cases would likely suffer from warning users even when the overriding they're doing is intentional -- the `AbstractUser` case I provided above is a prime example of one that would result in warnings that the user wouldn't be able to do anything about, because the parent class is defined in Django itself..